test/yargs.js

```diff
@@ -301,7 +301,26 @@ describe('yargs dsl tests', () => {
         .exitProcess(false) // defaults to true.
         .argv
     })
-
+    it('run middlewares before reaching the handler', function (done) {
+      yargs(['foo'])
+        .command(
+          'foo',
+          'handle foo things',
+          function () {},
+          function (argv) {
+            // we should get the argv filled with data from the middleware
+            argv._[0].should.equal('foo')
+            console.log(argv)
+            argv.hello.should.equal('world')
+            return done()
+          },
+        [function (argv) {
+          return {hello: 'world'}
```

Comment:
slick, so middleware is additive? we should definitely make sure we mention that it can both add new keys (while maintaining the existing) or overwrite old keys.